libs/watt/src/lib/components/picker/datepicker/watt-datepicker-v2.component.ts

```diff
@@ -0,0 +1,414 @@
+
+  /**
+   * @ignore
+   */
+  @ViewChild('endDateInput')
+  endInput!: ElementRef;
+  /**
+   * @ignore
+   */
+  protected _placeholder = this.getPlaceholder(this.getInputFormat());
+
+  @Input() dateClass: MatCalendarCellClassFunction<Date> = () => '';
+
+  /**
+   * @ignore
+   */
+  datePlaceholder = this.getPlaceholderByLocale(this.locale);
+  /**
+   * @ignore
+   */
+  rangeSeparator = ' - ';
+  /**
+   * @ignore
+   */
+  rangePlaceholder = this.datePlaceholder + this.rangeSeparator + this.datePlaceholder;
+  /**
+   * @ignore
+   */
+  inputMask: MaskitoOptions = maskitoDateOptionsGenerator({ mode: 'dd/mm/yyyy', separator: '-' });
+  /**
+   * @ignore
+   */
+  rangeInputMask: MaskitoOptions = maskitoDateRangeOptionsGenerator({
+    mode: 'dd/mm/yyyy',
+    dateSeparator: '-',
+  });
+  /**
+   * @ignore
+   */
+  getPlaceholderByLocale(locale: WattSupportedLocales): string {
+    return locale === 'da' ? 'dd-mm-åååå' : 'dd-mm-yyyy';
+  }
+  getRangePlaceholder(): string {
+    return this.datePlaceholder + this.rangeSeparator + this.datePlaceholder;
+  }
+  constructor() {
+    super(`watt-datepicker-v2-${WattDatepickerV2Component.nextId++}`);
+    this.localeService.onLocaleChange$
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((locale) => {
+        this.datePlaceholder = this.getPlaceholderByLocale(locale);
+        this.rangePlaceholder = this.getRangePlaceholder();
+      });
+  }
+
+  protected initSingleInput() {
+    if (this.initialValue) {
+      this.matDatepickerInput.value = this.initialValue;
+      this.datepickerClosed();
+    }
+  }
+
+  inputChanged(value: string) {
+    const dateString = value.slice(0, this.datePlaceholder.length);
+    if (dateString.length === 0) {
+      this.control?.setValue(null);
+      return;
+    }
+    if (dateString.length !== this.datePlaceholder.length) {
+      return;
+    }
+    const date = this.parseDateShortFormat(dateString);
+    this.control?.setValue(this.formatDateFromViewToModel(date));
+  }
+
+  datepickerClosed() {
+    if (this.matDatepickerInput.value) {
+      this.control?.setValue(this.matDatepickerInput.value);
+      (this.actualInput.nativeElement as HTMLInputElement).value =
+        this.formatDateTimeFromModelToView(
+          this.formatDateFromViewToModel(this.matDatepickerInput.value)
+        );
+    } else {
+      (this.actualInput.nativeElement as HTMLInputElement).value = '';
+      this.control?.setValue(null);
+    }
+    this.actualInput.nativeElement.dispatchEvent(new InputEvent('input'));
+  }
+
+  onMonthSelected(date: Date) {
+    if (this.rangeMonthOnlyMode && date) {
+      this.matDateRangePicker.select(startOfMonth(date));
+      this.matDateRangePicker.select(endOfMonth(date));
+      this.matDateRangePicker.close();
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  protected initRangeInput() {
+    if (this.initialValue) {
+      this.matStartDate.value = (this.initialValue as WattDateRange).start;
```

Comment:
Suggestion: Do not use a deprecated type in a new version

Reply:
It's probably quite a big change as this stems from the WattPickerBase which should then be rewritten or fully excluded from the new implementation. I'd rather keep this as is and the change it when we remove the old date/time components?